packages/react-compose/src/mergeShorthandProp.ts

```diff
@@ -0,0 +1,19 @@
+import { ShorthandValue } from './types';
+import * as React from 'react';
+
+export function mergeShorthandProp<TProps>(
```

Comment:
Why is this needed? Isn't this already done in mergeProps?

Reply:
or is the idea to move it out of mergeProps?

Reply:
i think we need to have the logic in mergeProps along with this helper. see the use case in `useCheckbox`

Reply:
I think Xu's using it as a middle ground because she's not yet moved the state to use `mergeProps`.

Reply:
@khmakoto , this is not a middle ground. this is something we need and user who author `state` would need. this needs to live along with `mergeProps`  (and it's already used for `Checkbox`)

Reply:
@dzearing - Updated code. now I'm using this method in `resolveSlotProps` to remove duplication